### PR TITLE
State transfer throughput window size configuration option

### PR DESCRIPTION
**Pull request by mdarade**
Configuration option to control logging interval of
state transfer's throughput details.

### CODE REVIEW

File: bftengine/include/bcstatetransfer/SimpleBCStateTransfer.hpp [hunk 1]
```diff
@@ -167,6 +167,7 @@ struct Config {
   bool runInSeparateThread = false;
   bool enableReservedPages = true;
   bool enableSourceBlocksPreFetch = true;
+  uint32_t getMissingBlocksSummaryWindowSize = 600;
```
Comment (glevkovich) [inline bftengine/include/bcstatetransfer/SimpleBCStateTransfer.hpp:170]: For the above:
The name I chose a long time ago is not accurate.
Can we change it to gettingMissingBlocksSummaryWindowSize?
nit: consider moving this under // sizes

For this whole file:
To be able to see the parameter printed, add it to operator<<() function below, you missed it.

File: bftengine/src/bcstatetransfer/BCStateTran.cpp [hunk 2]
```diff
@@ -268,6 +268,8 @@ BCStateTran::BCStateTran(const Config &config, IAppState *const stateApi, DataSt
   // callbacks.
   addOnTransferringCompleteCallback(
       [this](uint64_t checkpoint_num) { replicaForStateTransfer_->onTransferringComplete(checkpoint_num); });
+
+  LOG_INFO(getLogger(), "ST throughput measurement " << KVLOG(config_.getMissingBlocksSummaryWindowSize));
```
Comment (glevkovich) [inline bftengine/src/bcstatetransfer/BCStateTran.cpp:272]: This is not needed, we already print the whole config by implementing config operator<<().
I think it should be removed.

File: kvbc/src/Replica.cpp [hunk 3]
```diff
@@ -398,6 +399,11 @@ Replica::Replica(ICommunication *comm,
   }
 #endif
 
+  if (stConfig.getMissingBlocksSummaryWindowSize > 0 and stConfig.getMissingBlocksSummaryWindowSize < 99) {
```
Comment (glevkovich) [inline kvbc/src/Replica.cpp:402]: < 100
or <= 99

**Review by glevkovich — CHANGES_REQUESTED** [review]
@mdarade, few small comments in the body.
Also, you forgot to modify bcstatetransfer_tests.cpp
See line (approx) 39:
`Config TestConfig() {`

Need to add the new parameter.

**Review by f-squirrel — APPROVED** [review]
LGTM with comments from @glevkovich .

**Review by glevkovich — APPROVED** [review]
LGTM. thanks!

**Pull request merged**